### DISCUSSION TITLE
Loopback ports into same component

### DISCUSSION THREAD

**Discussion by satzguyz** [nasa/fprime#1393] [Q&A]
Hello

Can I loopback the input and output port of the same component or is there any constraint on doing so ?

**Comment by timcanham** [comment]
If the output and input type of the ports are the same, there is nothing preventing you from loop back the connection. If you want to do that just for internal connections, consider using `internal port` like here: 

https://github.com/nasa/fprime/blob/devel/Svc/ActiveLogger/ActiveLogger.fpp#L34

That allows you to avoid creating all the port code and invoking it, since internally the `internal port` will just queue a message.
